### PR TITLE
Docs: Fix broken links

### PR DESCRIPTION
**Pull request by dgault**
Fixing 4 broken documentation links: 
https://ci.openmicroscopy.org/view/Failing/job/BIOFORMATS-DEV-merge-docs/766/warnings3Result/new/

To test:
- Ensure builds are green and links are valid

### CODE REVIEW

**Comment by melissalinkert** [comment]
Could the corresponding links in https://github.com/openmicroscopy/bioformats/blob/develop/components/autogen/src/format-pages.txt be updated as well please?

**Comment by rleigh-codelibre** [comment]
The updated links look fine to me.

File: components/autogen/src/format-pages.txt [hunk 1]
```diff
@@ -956,10 +956,10 @@ reader = ImprovisionTiffReader
 [Imspector OBF]
 extensions = .obf, .msr
 owner = `MPI-BPC <http://www.mpibpc.mpg.de/>`_
-developer = `Department of NanoBiophotonics, MPI-BPC <https://imspector.mpibpc.mpg.de/index.html>`_
+developer = `Department of NanoBiophotonics, MPI-BPC <https://imspector.mpibpc.mpg.de>`_
 bsd = yes
 weHave = * a few .msr datasets \n
-* `a specification document <https://imspector.mpibpc.mpg.de/documentation/fileformat.html>`_
+* `a specification document <https://imspectordocs.readthedocs.io/en/latest/fileformat.html#>`_
```
Comment (rleigh-codelibre) [inline components/autogen/src/format-pages.txt]: Is the trailing `#` needed here (same for the other instance of this link)?

**Comment by rleigh-codelibre** [comment]
The latest changes look fine to me.  Just one question about the `#` in the link (which works with or without, it seems).

File: components/autogen/src/format-pages.txt (hunk 1 above)
Comment (sbesson) [inline components/autogen/src/format-pages.txt]: I agree the trailing `#` is not necessary and should be dropped from this PR.

**Pull request merged**